controller/auth/manager.py

```diff
@@ -152,3 +155,51 @@ def extract_state_info(request: Request, key: str) -> Any:
         return value
 
     return request.state.parsed[key]
+
+
+def check_is_full_admin(request: Any) -> bool:
+    if check_is_admin(request):
+        jwt_decoded: Dict[str, Any] = jwt.decode(
+            request.headers["Authorization"].split(" ")[1],
+            options={"verify_signature": False},
+        )
+        subject: Dict[str, Any] = jwt_decoded["session"]["identity"]
+        if check_email_in_full_admin(subject["traits"]["email"]):
+            return True
+    return False
+
+
+def invite_users(
+    emails: List[str], organization_name: str, provider: Optional[str] = None
+):
+    for email in emails:
+        # Create accounts for the email
+        user = kratos.create_user_kratos(email, provider)
+        if not user:
+            raise AuthManagerError("User creation failed")
+
+        # Assign the account to the organization
+        user_manager.update_organization_of_user(organization_name, email)
+
+        # Get the recovery link for the email
+        recovery_link = kratos.get_recovery_link(user["id"])
+        if not recovery_link:
+            raise AuthManagerError("Failed to get recovery link")
+
+        # Send the recovery link to the email
+        kratos.email_with_link(email, recovery_link["recovery_link"])
+
+
+def check_valid_emails(emails: List[str]):
+
+    for email in emails:
+        if not is_valid_email(email):
+            return False
+        if kratos.check_user_exists(email):
+            return False
+    return True
+
+
+def is_valid_email(email):
+    pattern = r"^[\w\.-]+@[\w\.-]+\.\w+$"
```

Comment:
this creates the pattern on every call of is_valid_email.

Since the regex doesn't change it's better to pre compile it outside the function scope & use fullmatch so it's not jsut a part of the string something like:

```
import re

EMAIL_RE = re.compile(r"[\w\.-]+@[\w\.-]+\.\w+")

def is_valid_email(email: str) -> bool:
    return bool(EMAIL_RE.fullmatch(email))
```